src/components/comments/comments.tsx

```diff
@@ -0,0 +1,33 @@
+import "../singleNewsPage/singleNewsPage.scss"
+import ChildComment from "../childComment/childComment";
+import {useState} from "react";
+import {dateServices} from "../../services/dateServices";
+import {IComment} from "../../models/INews";
+
+type Props = {
+    comment: IComment;
+}
+
+const Comments = ({comment}: Props) => {
+
+    const [visible, setVisible] = useState(false)
+
+    return (
+        <div onClick={() => comment?.kids && setVisible(!visible)}>
+            <li
+                className="single-news__comment__item"
+                key={comment.id}>
+                <p>{comment.by} {dateServices(comment.time)}</p>
+                <p>{comment.text}</p>
+                <p>Count child comments {"kids" in comment ? comment.kids.length : 0}</p>
```

Comment:
Такую штуку всегда можно сократить с использованием optional chaining
```suggestion
                <p>Count child comments {comment.kids?.length ?? 0}</p>
```